CMakeLists.txt

```diff
@@ -1070,6 +1069,7 @@ if(WITH_TESTS)
   add_custom_target(check COMMAND ${CMAKE_CTEST_COMMAND})
   set(TESTUTILLIB testutillib${ARTIFACT_SUFFIX})
   add_library(${TESTUTILLIB} STATIC ${TESTUTIL_SOURCE})
+  target_link_libraries(${TESTUTILLIB} ${LIBS})
```

Comment:
@merryChris this makes sense, as some source files (`db/db_test_util.h` for instance) in `${TESTUTILLIB}` actually references headers used by `ROCKSDB_STATIC_LIB/ROCKSDB_SHARED_LIB`, but  i'd suggest move the cmake changes into a separated commit. and explain the reason why we need this change in the commit message.

Reply:
Fine, I will pick up the cmake changes in a separated commit. Close this pr temporarily.

Reply:
@merryChris JFYI, there is no need to create a new PR for addressing comments in existing one. you can always `git push -f` to your remote branch for updating a PR.

Reply:
@tchaikov thanks for reminding.

Reply:
@tchaikov I am not sure why I cannot update PR automatically. Sorry for that, and plz have a review at https://github.com/facebook/rocksdb/pull/6045.